nixpkgs_review/github.py

```diff
@@ -26,37 +38,67 @@ def _request(
             body = json.dumps(data).encode("ascii")
 
         req = urllib.request.Request(url, headers=headers, method=method, data=body)
-        resp = urllib.request.urlopen(req)
+        try:
+            resp = urllib.request.urlopen(req)
+        except urllib.error.HTTPError as e:
+            print(f"Url: {url}", file=sys.stderr)
+            print(f"Code: {e.code}", file=sys.stderr)
+            print(f"Reason: {e.reason}", file=sys.stderr)
+            print(f"Headers: {e.headers}", file=sys.stderr)
+            print(f"Request data: {data}", file=sys.stderr)
+            print(f"Response: {e.read().decode('utf-8')}", file=sys.stderr)
+            raise
+
         return json.loads(resp.read())
 
     def get(self, path: str) -> Any:
         return self._request(path, "GET")
 
-    def post(self, path: str, data: Dict[str, str]) -> Any:
+    def post(self, path: str, data: Dict[str, Any]) -> Any:
```

Comment:
Can we be more specific than Any?

Reply:
Not really. You can post any JSON data to the API. The gist API requires some nested dicts, which is the reason for the change.